src/ussd_popi_user_data.js

```diff
@@ -0,0 +1,445 @@
+                case 'nso_ZA':
+                    return 'Sesotho sa Leboa';
+                case 'tsn_ZA':
+                    return 'Setswana';
+                case 'sot_ZA':
+                    return 'Sesotho';
+                case 'tso_ZA':
+                    return 'Xitsonga';
+                case 'ssw_ZA':
+                    return 'siSwati';
+                case 'ven_ZA':
+                    return 'Tshivenda';
+                case 'nbl_ZA':
+                    return 'isiNdebele';
+                }
+        };
+
+        // override normal state adding
+        self.add = function(name, creator) {
+            self.states.add(name, function(name, opts) {
+                if (!interrupt || !utils.timed_out(self.im))
+                    return creator(name, opts);
+
+                interrupt = false;
+                var timeout_opts = opts || {};
+                timeout_opts.name = name;
+                return self.states.create('state_timed_out', timeout_opts);
+            });
+        };
+
+        self.states.add('state_timed_out', function(name, creator_opts) {
+            return new ChoiceState(name, {
+                question: $('Welcome back. Please select an option:'),
+                choices: [
+                    new Choice('state_start', $('Main menu'))
+                ],
+                next: function(choice) {
+                    return choice.value;
+                }
+            });
+        });
+
+
+        self.add("state_start", function(name) {
+            self.im.user.set_answers = {};
+            var msisdn = utils.normalize_msisdn(self.im.user.addr, '27');
+            self.im.user.set_answer("operator_msisdn", msisdn);
+
+            return is
+            .get_or_create_identity({"msisdn": msisdn})
+            .then(function(identity) {
+                self.im.user.set_answer("operator", identity);
+                self.im.user.set_answer("msisdn",msisdn);
+                
+                // display in previously chosen language
+                self.im.user.set_lang(self.im.user.answers.state_language);
+                
+                return sbm
+                // check that user is registered on momconnect   
+                .check_identity_subscribed(self.im.user.answers.operator.id, "momconnect")
+                .then(function(identity_subscribed_to_momconnect) {
+                    if (identity_subscribed_to_momconnect) {
+                        var promises = [];
+                        return sbm
+                        .list_active_subscriptions(self.im.user.answers.operator.id)
+                        .then(function(active_subscriptions){
+                            promises = active_subscriptions.results.map(function(result){
+                                return sbm.get_messageset(result.messageset); 
+                            });
+                            var sets = '';
+                            return Promise.all(promises)
```

Comment:
Because the javascript sandbox was written in an older version of node, and still runs using that older version, `Promise` didn't exist, so we used the `Q` library, which has the same functions: http://documentup.com/kriskowal/q/#combination . I think we can use that instead of having to include the extra dependancy.

Sorry for the confusion around this.